walk/walk.go

```diff
@@ -115,7 +115,7 @@ func Walk(c *config.Config, cexts []config.Configurer, dirs []string, mode Mode,
 		}
 	}
 
-	updateRels := buildUpdateRelMap(c.RepoRoot, dirs)
+	updateRels := NewUpdateFilter(c.RepoRoot, dirs, mode)
```

Comment:
I could potentially pre-factor all this?

It's basically just wrapping all the util methods into a struct. That struct is only used internally but needs to be "public" (the capital `New`) because it's used in both `walk/` and `resolve/`.

Reply:
https://github.com/bazelbuild/bazel-gazelle/pull/1881 if we want to merge that first